dlib/geometry/rectangle.h

```diff
@@ -750,8 +750,10 @@ namespace dlib
 
     // Circumvent what appears to be a bug in Visual Studio 2019's optimizer
     // (see: https://forum.juce.com/t/warning-in-the-lastest-vs2019/38267)
+#if defined (_MSVC_VER)
```

Comment:
Actually it seems that this should be `_MSC_VER`, i.e. no `V` there.

I will include a fix in #2490.